README.md

```diff
@@ -6,65 +8,135 @@
 [![Code Quality: Java](https://img.shields.io/lgtm/grade/java/g/hazelcast/hazelcast.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/hazelcast/hazelcast/context:java)
 [![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=hz-os-master&metric=alert_status)](https://sonarcloud.io/dashboard?id=hz-os-master)
 
-## Hazelcast is a clustering and highly scalable data distribution platform.
+----
+
+Hazelcast is an open-source distributed in-memory data store and
+computation platform. It provides a wide variety of distributed data
+structures and concurrency primitives, including:
+
+* a distributed, partitioned and queryable in-memory key-value store
+  implementation, called `IMap`
+* additional data structures and simple messaging constructs such as
+  `Set`, `MultiMap`, `Queue`, `Topic`
+* cluster-wide unique ID generator, called `FlakeIdGenerator`
+* a distributed, [CRDT](https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type)
+  based counter, called `PNCounter`
+* a cardinality estimator based on [`HyperLogLog`](https://en.wikipedia.org/wiki/HyperLogLog)
+
+Additionally, Hazelcast includes a production-ready
+[Raft](https://en.wikipedia.org/wiki/Raft_(computer_science))
+implementation which allows implementation of _linearizable_ constructs
+such as:
+
+* a distributed and reentrant lock implementation, called `FencedLock`
+* primitives for distributed computing such as `AtomicLong`,
+`AtomicReference` and `CountDownLatch`
```

Comment:
Let's put a period at the end of last bulleted item since the above list is a phrase of the sentence starting with "Additionally".